lib/src/specs/expression.dart

```diff
@@ -65,27 +139,74 @@ abstract class Expression implements Spec {
   Expression newInstance(
     List<Expression> positionalArguments, [
     Map<String, Expression> namedArguments = const {},
+    List<Reference> typeArguments = const [],
   ]) {
     return new InvokeExpression._new(
       this,
       positionalArguments,
       namedArguments,
+      typeArguments,
+      null,
+    );
+  }
+
+  /// Returns a new instance of this expression with a named constructor.
+  Expression newInstanceNamed(
+    String name,
+    List<Expression> positionalArguments, [
+    Map<String, Expression> namedArguments = const {},
+    List<Reference> typeArguments = const [],
+  ]) {
+    return new InvokeExpression._new(
+      this,
+      positionalArguments,
+      namedArguments,
+      typeArguments,
+      name,
     );
   }
 
   /// Returns a const instance of this expression.
   Expression constInstance(
     List<Expression> positionalArguments, [
     Map<String, Expression> namedArguments = const {},
+    List<Reference> typeArguments = const [],
   ]) {
     return new InvokeExpression._const(
       this,
       positionalArguments,
       namedArguments,
+      typeArguments,
+      null,
     );
   }
 
-  /// May be overriden to support other types implementing [Expression].
+  /// Returns a const instance of this expression with a named constructor.
```

Comment:
I'm not sure what this does. Could you provide an example (and add it to the dartdoc)? Specifically, I'm not aware of many kinds of expressions that you can _invoke_ with a named constructor, i.e. only references to classes. Perhaps this method should be moved to `Reference`?